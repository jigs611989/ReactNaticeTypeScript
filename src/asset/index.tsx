export { default as Color } from './Color';
export { default as AppStyle } from './AppStyle';
export { default as String } from './String';
export { default as Images } from './image';
