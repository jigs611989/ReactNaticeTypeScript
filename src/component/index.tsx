export { default as Button } from './Button';
export { default as Label } from './Label';
export { default as UserName } from './UserName';
export { default as TextField } from './TextField';
export { default as SliderButton } from './SliderButton';
