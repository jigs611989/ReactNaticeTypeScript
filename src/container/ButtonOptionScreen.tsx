import React from 'react';
import { View, Text } from 'react-native';

export interface Props {}

const ButtonOptionScreen: React.FC<Props> = (props) => {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Text>Button Option Screen</Text>
    </View>
  );
};

export default ButtonOptionScreen;
