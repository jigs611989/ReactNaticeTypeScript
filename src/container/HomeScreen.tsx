import React from 'react';
import { View, Text, Pressable } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Screens from '../constant/Screens';

export interface Props {}

const HomeScreen: React.FC<Props> = (props) => {
  const { navigate } = useNavigation();
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Pressable onPress={() => navigate(Screens.ButtonOptionScreen)}>
        <Text>Home Screen</Text>
      </Pressable>
    </View>
  );
};

export default HomeScreen;
