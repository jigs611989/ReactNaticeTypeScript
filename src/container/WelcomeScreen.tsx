import React from 'react';
import { View, Text, Pressable } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Screens from '../constant/Screens';

export interface Props {}

const WelcomeScreen: React.FC<Props> = (props) => {
  const { navigate } = useNavigation();
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Pressable onPress={() => navigate(Screens.AppStack)}>
        <Text>Welcome Screen</Text>
      </Pressable>
    </View>
  );
};

export default WelcomeScreen;
