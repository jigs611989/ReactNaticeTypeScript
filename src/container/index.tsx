export { default as WelcomeScreen } from './WelcomeScreen';
export { default as HomeScreen } from './HomeScreen';
export { default as ButtonOptionScreen } from './ButtonOptionScreen';
