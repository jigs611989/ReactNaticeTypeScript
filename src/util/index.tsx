import { regex } from '../constant/AppConstant';
const isValidName = (name: string): boolean => {
  if (name.match(regex.name)) {
    return true;
  }
  return false;
};

export { isValidName };
