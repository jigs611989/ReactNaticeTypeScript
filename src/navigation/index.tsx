import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import HomeScreen from '../container/HomeScreen';
import WelcomeScreen from '../container/WelcomeScreen';
import ButtonOptionScreen from '../container/ButtonOptionScreen';
import Screens from '../constant/Screens';

const Stack = createStackNavigator();

function OnboardingStack() {
  return (
    <Stack.Navigator headerMode="none">
      <Stack.Screen name={Screens.WelcomeScreen} component={WelcomeScreen} />
    </Stack.Navigator>
  );
}

function AppStack() {
  return (
    <Stack.Navigator>
      <Stack.Screen name={Screens.HomeScreen} component={HomeScreen} />
      <Stack.Screen
        name={Screens.ButtonOptionScreen}
        component={ButtonOptionScreen}
      />
    </Stack.Navigator>
  );
}

function MainNavigation() {
  return (
    <NavigationContainer>
      <Stack.Navigator headerMode={'none'}>
        <Stack.Screen
          name={Screens.OnboardingStack}
          component={OnboardingStack}
        />
        <Stack.Screen name={Screens.AppStack} component={AppStack} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default MainNavigation;
