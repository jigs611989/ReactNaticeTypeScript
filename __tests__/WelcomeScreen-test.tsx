import React from 'react';
import { WelcomeScreen } from '../src/container';
import { render } from '@testing-library/react-native';

it('renders correctly', () => {
  const { toJSON } = render(<WelcomeScreen />);
  expect(toJSON()).toMatchSnapshot();
});
