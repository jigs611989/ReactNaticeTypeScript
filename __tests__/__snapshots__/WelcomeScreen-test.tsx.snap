// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text>
      Welcome Screen
    </Text>
  </View>
</View>
`;
